src/buffer_object.rs

```diff
@@ -319,8 +302,7 @@ impl<T: 'static> BufferObject<T> {
         D: AsFd + 'static,
         F: FnOnce(&MappedBufferObject<'a, T>) -> S,
     {
-        let device_ref = self._device.upgrade().ok_or(WrongDeviceError)?;
-        if *device_ref != device.as_raw_mut() {
+        if *self._device != device.as_raw_mut() {
```

Comment:
I wonder what safety guarantee this was meant to uphold anyway to borrow the device on map. The `Device` has no methods requiring a mutable reference, so this does nothing, but being annoying, especially now that we have a strong pointer anyway.

Lets just drop the `device` argument.